lib/spack/spack/build_environment.py

```diff
@@ -302,6 +302,19 @@ def set_build_environment_variables(pkg, env, dirty=False):
             if '/macports/' in p:
                 env.remove_path('PATH', p)
 
+    # Set environment variables if specified for
+    # the given compiler
+    compiler = pkg.compiler
+    environment = compiler.environment
+    if 'set' in environment:
+        env_to_set = environment['set']
+        for key, value in env_to_set.iteritems():
+            env.set('SPACK_%s' % key, value)
```

Comment:
I think this makes it possible for certain user variables to overwrite Spack's various `cc` control variables.  e.g., what happens if the user says (for who knows what reason) that `CC` should be set a certain way while the compiler runs -- it will conflict with `SPACK_CC`.  Should this use a different prefix, like `SPACK_ENV_SET_` or similar?